packages/protocol/contracts/identity/Attestations.sol

```diff
@@ -526,6 +546,59 @@ contract Attestations is IAttestations, Ownable, Initializable, UsingRegistry, R
     return accounts[account].dataEncryptionKey;
   }
 
+  /**
+   * @notice Setter for the attestation key for an account
+   * @param attestationKeyAddress The attestation key address to set for the account
+   * @param v The recovery id of the incoming ECDSA signature.
+   * @param r Output value r of the ECDSA signature.
+   * @param s Output value s of the ECDSA signature.
+   * @dev v, r, s constitute `attestationKeyAddress`'s signature on `msg.sender`.
+   */
+  function setAttestationKeyAddress(
```

Comment:
Should we impose any additional restrictions here? e.g. should two accounts be able to use the same attestations key?

Reply:
I was thinking briefly about it. They should not, but the same key could only map to one account. So we would have an inconsistency that if someone maps the attestation key to a different account, then the old account still specifies the old attestation key, but should still be able to sign with the account key.

We could wipe the authorizedAttestor of the old account when an attestation key gets changed, but it felt a bit weird to me that the new account could impact the struct of the other account. What do you think?

Reply:
I think we should do the same thing we do in locked gold, where it is invalid for an address to be both an account and an authorized validator/voter, and where it is invalid for two accounts to authorize the same address